apps/client/src/features/app-settings/panel/general-panel/StyleEditor.tsx

```diff
@@ -0,0 +1,43 @@
+import { useEffect, useState } from 'react';
+import Editor from 'react-simple-code-editor';
+import Prism from 'prismjs';
+
+import 'prismjs/components/prism-json';
+
+import 'prismjs/components/prism-css';
+import 'prismjs/themes/prism-funky.min.css';
+import style from './StyleEditor.module.scss';
+
+interface CodeEditorProps {
+  language: string;
+  initialValue: string;
+  onChange: (newValue: string) => void;
+}
+
+const CodeEditor = (props: CodeEditorProps) => {
```

Comment:
questions: 
- we usually use the function keyword and export default inline, is there a reason not to do this here?
- is this component a good candidate for memoisation? it sounds like it can be slow to render